editoast/src/views/timetable.rs

```diff
@@ -933,4 +989,126 @@ mod tests {
             .json_into();
         assert_eq!(&response.error_type, "editoast:timetable:NotFound")
     }
+
+    #[test]
+    fn train_id_parse() {
```

Comment:
There is 4 different test in this function. So I would rather make 4 test functions than one. Note that you might be able to leverage `#[rstest]` and `#[case]` to write the 4 tests in the same function 🤷